File: leetcode-169-Majority_Element/leetcode-169-Majority_Element.py
def majorityElement(nums: list[int]) -> int:
        coount_apearance = {}
        n_2 = len(nums)//2
        for x in nums:
            if coount_apearance.get(x) == None:
                coount_apearance[x] = 0
            coount_apearance[x] += 1
            if coount_apearance[x] > n_2:
                return x
        return -1


if __name__ == "__main__":
    vec = [2,2,1,1,1,2,2]
    res = majorityElement(vec)
    print(res)

File: leetcode-169-Majority_Element/leetcode-169-Majority_Element.cpp
#include <iostream>
#include <cstdlib>
#include <vector>
#include <unordered_map>


class Solution {
public:
    int majorityElement(std::vector<int>& nums) {
        std::unordered_map<int, int> vals_count;
        size_t half_size = nums.size()/2;
        
        for(const auto& x:nums)
        {
            if(half_size<(++vals_count[x]))
            {
                return x;
            }
        }
        return 0;
    }
};

int main() {
    std::vector<int> nums = {3,2,3};
    
    auto res = Solution().majorityElement(nums);

    std::cout << res << std::endl;
    
    return EXIT_SUCCESS;
}